src/Applications/RevitTestFrameworkGUI/TextBoxOutputter.cs

```diff
@@ -2,25 +2,61 @@
 using System.IO;
 using System.Text;
 using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
 
 namespace RTF.Applications
 {
     public class TextBoxOutputter : TextWriter
     {
-        TextBox textBox = null;
+        RichTextBox textBox = null;
+        StringBuilder pendingText = new StringBuilder();
 
-        public TextBoxOutputter(TextBox output)
+        public TextBoxOutputter(RichTextBox output)
         {
             textBox = output;
         }
 
         public override void Write(char value)
         {
             base.Write(value);
-            textBox.Dispatcher.BeginInvoke(new Action(() =>
+            if (value != '\n')
             {
-                textBox.AppendText(value.ToString());
-            }));
+                pendingText.Append(value);
+
+                if (value == '\r')
+                {
+                    string line = pendingText.ToString();
+                    pendingText.Clear();
+
+                    textBox.Dispatcher.BeginInvoke(new Action(() =>
+                    {
+                        TextRange tr = new TextRange(textBox.Document.ContentEnd, textBox.Document.ContentEnd);
+                        tr.Text = $"{DateTime.Now.ToString("HH:mm:ss")} ";
+                        tr.ApplyPropertyValue(TextElement.ForegroundProperty, Brushes.DarkGray);
+
+                        tr = new TextRange(textBox.Document.ContentEnd, textBox.Document.ContentEnd);
+                        tr.Text = line;
+                        if (line.ToLower().Contains("error:"))
```

Comment:
a small thing - would be nice if these could be referred to throughout the code using an enum or const.

Reply:
good point, thanks! will add consts